src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -13772,6 +13772,28 @@ instruct leaL_rReg_immI2_peep(rRegL dst, rRegL src, immI2 shift)
   ins_pipe(ialu_reg_reg);
 %}
 
+instruct testI_reg_empty_peep(rFlagsReg cr, rRegI src, immI_0 zero)
+%{
+  predicate(false);
+  match(Set cr (CmpI src zero));
+
+  format %{ "# noop testI - cr set by and" %}
+  ins_encode %{
+  %}
+  ins_pipe(ialu_reg);
+%}
+
+instruct testL_reg_empty_peep(rFlagsReg cr, rRegL src, immL0 zero)
+%{
+  predicate(false);
+  match(Set cr (CmpL src zero));
+
+  format %{ "# noop testL - cr set by and" %}
+  ins_encode %{
+  %}
+  ins_pipe(ialu_reg);
+%}
+
```

Comment:
In `test_may_remove_5` the `new_root` function points to the constructor of one of these instructions (what's passed to `peepreplace`). Since the function pointer is unused, these are really not needed. (for `peepreplace` you can just pass `testI_reg` or `testL_reg`, it doesn't really matter).

Reply:
Potentially we could modify ADLC to make `peepreplace` optional, and then pass `nullptr` to the peepprocedure.